eating_cookies/eating_cookies.py

```diff
@@ -3,14 +3,21 @@
 import sys
 
 # The cache parameter is here for if you want to implement
-# a solution that is more efficient than the naive 
+# a solution that is more efficient than the naive
 # recursive solution
-def eating_cookies(n, cache=None):
-  pass
+
+
+def eating_cookies(n, cache=[1, 1, 2]):
+    if len(cache) > n:
+        return cache[n]
+    cache.append(sum(cache[-3:]))
+    return eating_cookies(n, cache)
+
```

Comment:
very cool; took me a second to understand but a very nice recursive implementation